### PR TITLE
Update data for Rhino 1.7

### LINKED ISSUE #1687: Unmark rhino as obsolete and update compatibiliy for recent versions
Since Nashorn has been deprecated as Java's Javascript engine, Rhino now has increased importance for people who don't want their existing applications to rely on 20mb of GraalVM.

Additionally, Rhino as currently tabulated is very out of date. Rhino themselves maintain a [kangax-compatible](http://mozilla.github.io/rhino/compat/engines.html) table.

### PR DESCRIPTION
**Pull request by gausie**
Fixes #1687

* Wrote a test runner for Rhino 1.7 (based off the existing nashorn runner)
* Fixed all (1000+ :sweat_smile:) out-of-date warnings in the data files

### CODE REVIEW

File: data-esnext.js [hunk 1]
```diff
@@ -124,6 +127,7 @@ exports.tests = [
         graalvm19: false,
         graalvm20: false,
         graalvm20_1: false,
+    rhino1_7: false,
```
Comment (ljharb) [inline data-esnext.js]: indentation needs a bit of polish

File: data-es2016plus.js [hunk 2]
```diff
@@ -408,13 +418,14 @@ exports.tests = [
...
           hermes0_7_0: true,
+          rhino1_7: true,
         }
       },
       {
         name: 'String.prototype.padEnd',
         spec: 'https://tc39.github.io/ecma262/#sec-string.prototype.padend',
         mdn: 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padEnd',
-        exec: function(){/*
+        exec: function () {/*
```
Comment (ljharb) [inline data-es2016plus.js]: let's revert these kinds of changes unless we have eslint enforcing it

File: data-es2016plus.js [hunk 3]
```diff
@@ -1156,8 +1187,8 @@ exports.tests = [
           firefox57: { val: "flagged", note_id: 'fx-shared-memory-spectre' },
           firefox72: firefox.nightly,
           firefox74: firefox.beta,
-          firefox77: {val: false, note_id: "fx-shared-memory-esr-disable" },
-          firefox79: {val: true, note_id: "fx-shared-memory-cors-isolation" },
+          firefox77: { val: false, note_id: "fx-shared-memory-esr-disable" },
+          firefox79: { val: true, note_id: "fx-shared-memory-cors-isolation" },
```
Comment (ljharb) [inline data-es2016plus.js]: also these

**Review by ljharb — CHANGES_REQUESTED** [review]
please revert unrelated whitespace/formatting changes

File: data-es2016plus.js (hunk 3 above)
Comment (gausie) [reply data-es2016plus.js]: Ah those are just my editor. I hate when it automatically formats things, so I don't know how it happened!

File: data-esnext.js [hunk 4]
```diff
@@ -1064,6 +1110,7 @@ exports.tests = [
         firefox10: false,
         firefox60: false,
         chrome77: false,
+        rhino1_7: false,
```
Comment (ljharb) [inline data-esnext.js]: all these trailing commas in the data are going to break older browsers. please ensure none of them have trailing commas.

**Review by ljharb — CHANGES_REQUESTED** [review]

File: data-esnext.js (hunk 4 above)
Comment (gausie) [reply data-esnext.js]: Okeydoke

Comment (gausie) [reply data-esnext.js]: @ljharb there are some other trailing commas throughout the files that I don't think I added, I'll remove them too.

Comment (gausie) [reply data-esnext.js]: I think these files are only consumed by `build.js` and aren't actually distributed via the web, but have made the change anyway

Comment (gausie) [reply data-esnext.js]: Ah no there were a few trailing commas in tests (that weren't testing trailing comma support). Not added by me, but I've removed them too. I get the feeling that I could have kept all these commas in though.

Comment (ljharb) [reply data-esnext.js]: thanks! it's important the site keeps working in old browsers.

**Review by ljharb — APPROVED** [review]

**Review by chicoxyzzy — APPROVED** [review]
Awesome, thanks!

**Pull request merged**